From 862566ffdf7a7b932f6f662fc63c977d1b8fab68 Mon Sep 17 00:00:00 2001
From: "Paul E. McKenney" <paulmck@linux.vnet.ibm.com>
Date: Mon, 27 Oct 2014 16:04:35 -0700
Subject: [PATCH] rcu: Fix FIXME in rcu_tasks_kthread()

commit 60ced4950c5059eff3f03027926eb5384f9923e1 upstream.

This commit affines rcu_tasks_kthread() to the housekeeping CPUs
in CONFIG_NO_HZ_FULL builds.  This is just a default, so systems
administrators are free to put this kthread somewhere else if they wish.

Signed-off-by: Paul E. McKenney <paulmck@linux.vnet.ibm.com>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

diff --git a/kernel/rcu/update.c b/kernel/rcu/update.c
index 97ed654c485e..bf2dcab80c58 100644
--- a/kernel/rcu/update.c
+++ b/kernel/rcu/update.c
@@ -533,7 +533,8 @@ static int __noreturn rcu_tasks_kthread(void *arg)
 	struct rcu_head *next;
 	LIST_HEAD(rcu_tasks_holdouts);
 
-	/* FIXME: Add housekeeping affinity. */
+	/* Run on housekeeping CPUs by default.  Sysadm can move if desired. */
+	housekeeping_affine(current);
 
 	/*
 	 * Each pass through the following loop makes one check for
-- 
2.3.4

